### PR TITLE
Added image compression

### PR DESCRIPTION
**Pull request by IEvangelist**
Hi @meganbradley,

I was working on this for a bit before I saw the "help-wanted" labels, so it might not have been on your radar. Nonetheless, it is really helpful. From the context menu, when selecting an image - you can choose to "Compress image". This will run a lossless image compression algorithm over many various image types:
- ".png"
- ".jpg"
- ".jpeg"
- ".gif"
- ".svg"
- ".webp"

**Before**
image

It also, exposes configurable max dimensions and will implicitly resize images before compression if required.

**Reported output**
image

### CODE REVIEW

**Comment by meganbradley** [comment]
Thanks @IEvangelist ! @lamebrain , please review!

**Comment by IEvangelist** [comment]
Updated this PR to add testing, and folder level compression as well.

**Folder level**

image

**Aggregated results**

image

**Comment by DuncanmaMSFT** [comment]
Just FYI @IEvangelist , this is great, I was just in a conversation with the team about writing something like this. Excited to get it reviewed and available

File: docs-authoring-pack/package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
   ],
   "extensionDependencies": [
     "DavidAnson.vscode-markdownlint",
+    "docsmsft.docs-images",
```
Comment (adunndevster) [inline docs-authoring-pack/package.json]: @IEvangelist - Can we just take this out for the moment? We have a CI/CD pipeline process we need to setup for this new extension. Once we get that setup, we will put this into the Docs Authoring Pack, and publish it to the marketplace. Sound good?

**Comment by adunndevster** [comment]
@IEvangelist - Beautiful work! I reviewed the code, and tested the extension locally and it does a great job compressing images with the ease of a right click :).

I just had one ask for the PR for now ^^. I'm going to work on the CI/CD publish pipeline for this so we can get it into the authoring pack.

File: docs-images/package.json [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        }
+      ]
+    },
+    "configuration": {
+      "type": "object",
+      "title": "Docs Images Extension Configuration",
+      "properties": {
+        "docsImages.maxWidth": {
+          "type": "number",
+          "default": 1200,
```
Comment (DuncanmaMSFT) [inline docs-images/package.json]: @adkinn Do you have a preference for the *default* setting for image maximum width in this Docs image compression extension? Most people won't change from the default, so we should pick the ideal Docs maximum as the default.

File: docs-authoring-pack/package.json (hunk 1 above)
Comment (IEvangelist) [reply docs-authoring-pack/package.json]: Yes, absolutely. I'll update it today.

File: docs-images/package.json (hunk 2 above)
Comment (IEvangelist) [reply docs-images/package.json]: Hi @DuncanmaMSFT,
I based this maximum on the platform suggested max width, called out in the [contributors guide](https://review.docs.microsoft.com/en-us/help/contribute/contribute-how-to-create-screenshot?branch=master#considerations). The max is only used when images are larger than they are recommended to be, and they will maintain the aspect ratio when automatically resized.

**Comment by IEvangelist** [comment]
Hi team, I've updated the pull request... rebased from master, and removed the `"docs-images"` in the main extension `package.json`.

**Comment by IEvangelist** [comment]
> @IEvangelist - Beautiful work! I reviewed the code, and tested the extension locally and it does a great job compressing images with the ease of a right click :).
> 
> I just had one ask for the PR for now ^^. I'm going to work on the CI/CD publish pipeline for this so we can get it into the authoring pack.

Hi @adunndevster,

The PR has been updated. Thank you for your kind words, they mean the world to me! Much appreciated... Happy holidays

**Pull request merged**